src/content/developers/tutorials/hello-world-smart-contract-fullstack/index.md

```diff
@@ -0,0 +1,1548 @@
+   //Smart contract events are a way for your contract to communicate that something happened on the blockchain to your app front-end, which can be 'listening' for certain events and take action when they happen.
+   event UpdatedMessages(string oldStr, string newStr);
+
+   // Declares a state variable `message` of type `string`.
+   // State variables are variables whose values are permanently stored in contract storage. The keyword `public` makes variables accessible from outside a contract and creates a function that other contracts or clients can call to access the value.
+   string public message;
+
+   // Similar to many class-based object-oriented languages, a constructor is a special function that is only executed upon contract creation.
+   // Constructors are used to initialize the contract's data. Learn more:https://solidity.readthedocs.io/en/v0.5.10/contracts.html#constructors
+   constructor(string memory initMessage) {
+
+      // Accepts a string argument `initMessage` and sets the value into the contract's `message` storage variable).
+      message = initMessage;
+   }
+
+   // A public function that accepts a string argument and updates the `message` storage variable.
+   function update(string memory newMessage) public {
+      string memory oldMsg = message;
+      message = newMessage;
+      emit UpdatedMessages(oldMsg, newMessage);
+   }
+}
+```
+
+This is a super simple smart contract that stores a message upon creation and can be updated by calling the `update` function.
+
+### Step 11: Connect Metamask & Alchemy to your project {#step-11-connect-metamask-&-alchemy-to-your-project}
+
+We’ve created a Metamask wallet, Alchemy account, and written our smart contract, now it’s time to connect the three.
+
+Every transaction sent from your virtual wallet requires a signature using your unique private key. To provide our program with this permission, we can safely store our private key (and Alchemy API key) in an environment file.
+
+> To learn more about sending transactions, check out [this tutorial](https://docs.alchemyapi.io/alchemy/tutorials/sending-transactions-using-web3-and-alchemy) on sending transactions using web3.
+
+First, install the dotenv package in your project directory:
+
+```
+npm install dotenv --save
+```
+
+Then, create a `.env` file in the root directory of our project, and add your Metamask private key and HTTP Alchemy API URL to it.
+
+Your environment file must be named `.env` or it won't be recognized as an environment file.
+
+Do not name it `process.env` or `.env-custom` or anything else.
+
+- Follow [these instructions](https://metamask.zendesk.com/hc/en-us/articles/360015289632-How-to-Export-an-Account-Private-Key) to export your private key
+- See below to get HTTP Alchemy API URL
+
+![](./get-alchemy-api-key.gif)
+
+Your `.env` should look like this:
+
+```
+API_URL = "https://eth-ropsten.alchemyapi.io/v2/your-api-key"
+PRIVATE_KEY = "your-metamask-private-key"
```

Comment:
Kinda think we should be clearer here. Last thing we want is people accidentally pushing their private keys to version control .